splunk/Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM docker.io/splunk/splunk:latest AS builder
+
+ENV SPLUNK_START_ARGS=--accept-license
+
+USER root
+
+COPY tests/test_logs /opt/splunk/test_logs
+COPY tests/indexes.conf /opt/splunk/etc/apps/search/local/indexes.conf
+COPY scripts/log_indexing.sh /opt/splunk/log_indexing.sh
+COPY scripts/pre-run.sh /opt/splunk/pre-run.sh
+
+RUN microdnf install -y jq && \
+    microdnf clean all
+
+USER ansible
+RUN bash /opt/splunk/pre-run.sh
+
+FROM docker.io/splunk/splunk:latest
+
+ENV SPLUNK_START_ARGS=--accept-license
+
+EXPOSE 8080
+EXPOSE 8089
+
+USER root
```

Comment:
you need to remember to change this back so the final image does not run as root.

also if you `COPY` as root you end up with root-owned files so you need to either properly chmod and chown them (IIRC there are flags to `COPY` to do that), or not `COPY` as  root to begin with.